Gems/ROS2/Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp

```diff
@@ -90,132 +89,294 @@ namespace ROS2
         return false;
     }
 
-    bool Utils::IsWheelURDFHeuristics(const urdf::LinkConstSharedPtr& link)
+    bool IsWheelURDFHeuristics(const sdf::Model& model, const sdf::Link* link)
     {
-        const AZStd::regex wheel_regex("wheel[_]||[_]wheel");
-        const AZStd::regex joint_regex("(?i)joint");
-        const AZStd::string link_name(link->name.c_str(), link->name.size());
+        const AZStd::regex wheelRegex("wheel[_]||[_]wheel");
+        const AZStd::regex jointRegex("(?i)joint");
+        const AZStd::string linkName(link->Name().c_str(), link->Name().size());
         AZStd::smatch match;
         // Check if name is catchy for wheel
-        if (!AZStd::regex_search(link_name, match, wheel_regex))
+        if (!AZStd::regex_search(linkName, match, wheelRegex))
         {
             return false;
         }
         // The name should contain a joint word
-        if (AZStd::regex_search(link_name, match, joint_regex))
+        if (AZStd::regex_search(linkName, match, jointRegex))
         {
             return false;
         }
```

Comment:
Actually I'll change this to use StartsWith and EndsWithd for the wheelRegex.

Reply:
I also just realized the `jointRegex` was actually returning true if the the joint name did contain "joint"
https://github.com/o3de/o3de-extras/blame/21affbdc4fc585395478d8eb587e8a9ba4b0b5d1/Gems/ROS2/Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp#L104-L108

Was this intended @michalpelka ?

Reply:
Also is there a need for the "joint" heuristic to filter out "wheel" links that contain the whose name also contained "joint"?

I imagine if the "wheel_" or "_wheel" heuristic above matched, then there is no need to check if the link name also contained the term "joint".

Especially since the element being checked is a strongly typed Link class in C++.